susemanager-utils/susemanager-sls/src/modules/mgractionchains.py

```diff
@@ -128,8 +145,16 @@ def start(actionchain_id):
     except Exception as exc:
         log.error("There was an error while syncing custom states and execution modules")
 
-    inside_transaction = os.environ.get("TRANSACTIONAL_UPDATE")
-    if __grains__.get("transactional") and not inside_transaction:
+    transactional_update = __grains__.get("transactional")
+    reboot_required = False
+    inside_transaction = False
+    if transactional_update:
+        reboot_required = check_reboot_required(target_sls)
+        inside_transaction = os.environ.get("TRANSACTIONAL_UPDATE")
+
+    if transactional_update and reboot_required:
+        ret = __salt__['transactional_update.sls'](target_sls, queue=True, activate_transaction=True)
+    elif transactional_update and not inside_transaction:
```

Comment:
Why is one of them checking `inside_transaction` but not the other? If both branches would do `if transactional_update and not inside_transaction [and reboot_required]`, they could be collapsed to:

```python
if transactional_update and not inside_transaction:
    ret = __salt__['transactional_update.sls'](target_sls, queue=True, activate_transaction=reboot_required)